tnc/audio.py

```diff
@@ -49,21 +64,33 @@ def fetch_audio_devices(input_devices, output_devices):
 
     """
     devices = sd.query_devices(device=None, kind=None)
-    for index, device in enumerate(devices):
+
+    # The use of set forces the list to contain only unique entries.
+    input_devs = set()
+    output_devs = set()
+
+    for device in devices:
         # Use a try/except block because Windows doesn't have an audio device range
         try:
             name = device["name"]
 
             max_output_channels = device["max_output_channels"]
             max_input_channels = device["max_input_channels"]
 
+        except KeyError:
+            continue
         except Exception as err:
             print(err)
             max_input_channels = 0
             max_output_channels = 0
             name = ""
 
         if max_input_channels > 0:
-            input_devices.append({"id": index, "name": name})
+            input_devs.add(device_crc(name))
```

Comment:
We should also include additional information like channels, supported sample rate and so on ( maybe all information which is stored in "devices"), so we can be sure we have a unique checksum, even if the names are equal ( happens with Icom ic705 for example. Same usb device name but different settings)

Reply:
Done. All data in the `device` structure is included in the checksum.